statsd/statsd.go

```diff
@@ -69,15 +69,36 @@ const (
 	defaultUDPPort      = "8125"
 )
 
+const (
+	// ddEntityID specifies client-side user-specified entity ID injection.
+	// This env var can be set to the Pod UID on Kubernetes via the downward API.
+	// Docs: https://docs.datadoghq.com/developers/dogstatsd/?tab=kubernetes#origin-detection-over-udp
+	ddEntityID = "DD_ENTITY_ID"
+
+	// ddEntityIDTag specifies the tag name for the client-side entity ID injection
+	// The Agent expects this tag to contain a non-prefixed Kubernetes Pod UID.
+	ddEntityIDTag = "dd.internal.entity_id"
+
+	// ddPrefixedOrigin specifies the tag name for the client-side prefixed entity ID injection.
+	// Currently the only supported format is dd.internal.prefixed_origin:container_id://<id>
+	// dd.internal.entity_id is prioritized over dd.internal.prefixed_origin for backward compatibility.
+	// If DD_ENTITY_ID is not set, the client tries to fill dd.internal.prefixed_origin automatically unless
+	// DD_UDP_ORIGIN_DISABLED is explicitly set to "true".
+	ddPrefixedOrigin = "dd.internal.prefixed_origin"
+
+	// udpOriginDisabled specifies the env var to disable injecting the dd.internal.prefixed_origin tag.
+	udpOriginDisabled = "DD_UDP_ORIGIN_DISABLED"
```

Comment:
This logic could be use in the same way for UDS. Could we rename this to remove the `UDP` part ? Something like `DD_ORIGIN_DETECTION` ?